### ISSUE TITLE
Error when trying 'MerlinTypeOf' at some types

### ISSUE THREAD

**Issue by AntouanK** [ocaml/merlin#617]
I'm having problems with merlin on VS Code, but I couldn't debug the issue, so I tried it on vim.
I tried the `MerlinTypeOf` at some types.
You can see that it works fine for some
 
<img width="798" alt="screen shot 2017-01-25 at 08 37 35" src="https://cloud.githubusercontent.com/assets/4569111/22283355/a84157ce-e2d9-11e6-86db-07749cf134a3.png">

And it throws an error on others ( this one comes from a module in another file, if that matters )

<img width="797" alt="screen shot 2017-01-25 at 08 36 14" src="https://cloud.githubusercontent.com/assets/4569111/22283363/ba695582-e2d9-11e6-843c-74deab32d280.png">

**Comment by let-def** [comment]
Seem to be related to the text encoding.
What is the error with VS Code?

Could you set the environment variable MERLIN_LOG=/tmp/merlin.log, rerun your test and share the /tmp/merlin.log file? It should help understanding.

**Comment by AntouanK** [comment]
I did set that variable, but when I launch VS Code, I don't see that file appear...
Is it a problem that I use zsh? I added the variable to .zshrc

**Comment by let-def** [comment]
Maybe VS Code doesn't inherit the environment. Try with vim from the same terminal :).

**Comment by AntouanK** [comment]
The images you see are in vim under zsh

**Comment by let-def** [comment]
Are there still no merlin.log when started from vim?

**Comment by AntouanK** [comment]
Had to restart the shell session...
Here you go: 
[merlin.log.txt](https://github.com/ocaml/merlin/files/729103/merlin.log.txt)
( I only opened vim, and tried MerlinTypeOf )

**Comment by let-def** [comment]
Ok I see. The cmi file contains an weird looking external declaration.
Am I right to assume that this `express` external was not handwritten but produced by Bucklescript rewritting?

**Comment by AntouanK** [comment]
Yes, I do use Bucklescript. I can push the files in a repo so you can see them.

**Comment by AntouanK** [comment]
Also, here's a log from opening VS Code in the same repo ( I just hoveres over some types, and tried Ctrl+space, althoug nothing comes up )
[merlin.log.2.txt](https://github.com/ocaml/merlin/files/729112/merlin.log.2.txt)

**Comment by let-def** [comment]
Thanks, it should be enough to diagnose the bug.

**Comment by let-def** [comment]
You can try with current git, it should work around the problem.

**Comment by AntouanK** [comment]
How do I install it?
Get the source, compile and point to the new binary?

**Comment by let-def** [comment]
`opam pin add merlin --dev-repo` should do it.

**Comment by AntouanK** [comment]
The error on vim is gone. But on VS Code, I still get no autocomplete or types on hover. I'll have to see what's wrong on VS Code.

Thank you.

**Comment by eimpec** [comment]
I have somewhat similar issue - js_of_ocaml ## seem to cause the problem
https://github.com/hackwaly/vscode-ocaml/issues/78#issuecomment-275145539
